hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/OMFailoverProxyProvider.java

```diff
@@ -552,5 +552,22 @@ protected void setProxiesForTesting(
     this.omProxyInfos = testOMProxyInfos;
     this.omNodeIDList = testOMNodeIDList;
   }
+
+  protected void setOmProxies(Map omProxies) {
+    this.omProxies = omProxies;
+  }
+
+  protected void setOmProxyInfos(Map omProxyInfos) {
+    this.omProxyInfos = omProxyInfos;
+  }
+
+  protected void setOmNodeIDList(List omNodeIDList) {
+    this.omNodeIDList = omNodeIDList;
+  }
```

Comment:
If duplication of `loadOMClientConfigs` is addressed, these would be unnecessary.  (Otherwise we can use existing `setProxiesForTesting` (and rename it, since it would no longer be only for testing)).

Reply:
See comments above for the Map data structures and related.  The `loadOMClientConfigs` method needed to be overridden in the `GrpcFailoverProxyProvider` subclass so to not initialize the `omProxyInfos` and omProxies parent classes with values not applicable to the Grpc failover (causes errors).  With the overridden method the parent class `omProxyInfos` and `omProxies` Map data structures are initialized with valid key sets that are necessary for `OmFailoverProxyProvider` methods used by the Grpc failover.

Reply:
Thanks @adoroszlai ; Using provided renamed `setProxiesForTesting` protected method in `OmFailoverProxyProvider` (renamed to `setProxies` and updated affected tests, `TestOMFailovers`).

Reply:
@adoroszlai ; pushed latest commit contains modifications made as commented.  Thanks.